src/translib/intf_app.go

```diff
@@ -396,8 +407,42 @@ func (app *IntfApp) doGetAllIpKeys(d *db.DB, dbSpec *db.TableSpec) ([]db.Key, er
 	return keys, err
 }
 
-/***********  Translation Helper fn to convert DB Interface info to Internal DS   ***********/
 
+func (app *IntfApp) getSpecificAttr(targetUriPath string, ifKey string, oc_val *ocbinds.OpenconfigInterfaces_Interfaces_Interface_State) (bool, error) {
+    switch targetUriPath {
+        case "/openconfig-interfaces:interfaces/interface/state/oper-status":
+            val,e := app.getIntfAttr(ifKey, PORT_OPER_STATUS  )
+            if len(val) > 0 {
+                switch val {
+                    case "up":
+                       oc_val.OperStatus = ocbinds.OpenconfigInterfaces_Interfaces_Interface_State_OperStatus_UP
+                    case "down":
+                       oc_val.OperStatus = ocbinds.OpenconfigInterfaces_Interfaces_Interface_State_OperStatus_DOWN
+                    default:
+                       oc_val.OperStatus = ocbinds.OpenconfigInterfaces_Interfaces_Interface_State_OperStatus_UNSET
+                }
+                return true,nil
+            }  else {
+                return true,e
+            }
+        default:
+            log.Infof("GET for " + targetUriPath + " not supported")
+    }
+    return false,nil
+}
+
+func (app *IntfApp) getIntfAttr(ifName string, attr string) (string, error) {
+
+    entry := app.ifTableMap[ifName]
```

Comment:
I guess, we need to check whether the key exists over here, otherwise it will throw an exception in the next statement. We can take care of it in next commit. Just a note down.